src/include/compat.h

```diff
@@ -38,6 +38,9 @@
 #undef ENODATA
 #define ENODATA ENOATTR
 #endif
+#ifndef EREMOTEIO
```

Comment:
could you consolidate this `#define` with the ones at line:79. so there is only one place that we define `EREMOTEIO` and `HOST_NAME_MAX`. also instead of `HOST_NAME_MAX `, FreeBSD has `MAXHOSTNAMELEN`. maybe we can define `HOST_NAME_MAX` like
```c
#ifndef HOST_NAME_MAX
# ifdef MAXHOSTNAMELEN)
#  define HOST_NAME_MAX MAXHOSTNAMELEN
# else
#  define HOST_NAME_MAX 64
# endif
#endif
```
BTW, 64 is a very conservative number. probably we can make it 255.